blog/_posts/2019-01-18-fluxdiffeq.md

```diff
@@ -0,0 +1,657 @@
+For the rabbits, let's say that we want to learn
+
+$$\text{rabbits tomorrow} = \text{Model}(\text{rabbits today}).$$
+
+In this case, we have prior knowledge of the rate of births being dependent on
+the current population. The way to mathematically state this
+structural assumption is via a differential equation. Here, what we are saying
+is that the birth rate of the rabbit population at a given time point increases
+when we have more rabbits. The simplest way of encoding that is
+
+$$\text{rabbits}'(t) = \alpha\cdot \text{rabbits}(t)$$
+
+where $\alpha$ is some learnable constant. If you know your calculus, the solution
+here is exponential growth from the starting point with a growth rate $\alpha$:
+$\text{rabbits}(t_\text{start})e^(\alpha t)$. But notice that we didn't need to know the
+solution to the differential equation to validate the idea: we encoded the
+structure of the model and mathematics itself then outputs what the solution
+should be. Because of this, differential equations have been the tool of choice
+in most science. For example, physical laws tell you how electrical quantities
+emit forces ([Maxwell's Equations](https://en.wikipedia.org/wiki/Maxwell%27s_equations)).
+These are essentially equations of how things change and thus
+"where things will be" is the solution to a differential equation. But in recent
+decades this application has gone much further, with fields like systems biology
+learning about cellular interactions by encoding known biological structures and
+mathematically enumerating our assumptions or in targeted drug dosage through
+PK/PD modelling in systems pharmacology.
+
+So as our machine learning models grow and are hungry for larger and larger
+amounts of data, differential equations have become an attractive option for
+specifying nonlinearities in a learnable (via the parameters) but constrained
+form. They are essentially a way of incorporating prior domain-specific knowledge of the structural relations
+between the inputs and outputs. Given this way of looking at the two, both methods
+tradeoff advantages and disadvantages, making them complementary tools for modeling.
+It seems like a clear next step in scientific practice to start putting them
+together in new and exciting ways!
+
+## What is the Neural Ordinary Differential Equation (ODE)?
+
+The neural ordinary differential equation is one of many ways to put these two
+subjects together. The simplest way of explaining it is that, instead of
+learning the nonlinear transformation directly, we wish to learn the structures
+of the nonlinear transformation. Thus instead of doing $y=ML(x)$, we put the
+machine learning model on the derivative, $y'(x) = ML(x)$, and now solve the ODE.
+Why would you ever do this? Well, one motivation is that, if you were to do this
+and then solve the ODE using the simplest and most error prone method, the
+Euler method, what you get is equivalent to a [residual neural network](https://arxiv.org/abs/1512.03385).

```

Comment:
Maybe it's because ODEs are my bread and butter, but I assumed it's pretty straightforward that if you see it as a method for solving some underlying ODE, you just write out that ODE and solve it another way. The "underlying ODE" part is the convergence of the Euler method as dt->0, which if others think should be mentioned in more detail please submit some text.

Reply:
It's not a question of implementation, but why we should see a model like ResNet as being like solving an ODE in the first place. Most people aren't going to come in with that perspective, or have seen that paper, so it's worth pointing out that we can see ResNet as approximating something "infinitely deep". The idea that deeper models are more powerful gives an intuition for why this gives good results.